angr/state_plugins/loop_data.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+from collections import defaultdict
+
+from .plugin import SimStatePlugin
+
+l = logging.getLogger('angr.state_plugins.loop_data')
+
+
+class SimStateLoopData(SimStatePlugin):
+    """
+    This class keeps track of loop-related information for states.
+    """
+
+    def __init__(self, trip_counts=None):
+        """
+        :param trip_counts: Dictionary that stores trip counts for each loop.
+        """
+
+        SimStatePlugin.__init__(self)
+
+        if trip_counts is None:
+            trip_counts = defaultdict(list)
+        self.trip_counts = trip_counts
+
+    def merge(self, others, merge_conditions, common_ancestor=None):
+        l.warning("Merging is not implemented for loop data!")
+        return False
+
+    def widen(self, others):
+        l.warning("Widening is not implemented for loop data!")
+        return False
+
+    def copy(self):
```

Comment:
It doesn't look like this will survive branching -- the trip_counts dict will be shared by both of the resulting plugins.

Reply:
Man, you're right. I completely forgot this :sob: